src/satosa/micro_services/ldap_attribute_store.py

```diff
@@ -303,6 +305,8 @@ def _ldap_connection_factory(self, config):
 
         pool_size = config["pool_size"]
         pool_keepalive = config["pool_keepalive"]
+        pool_name = ''.join(random.sample(string.ascii_lowercase, 6))
```

Comment:
I don't remember the entropy that `random` has; another idea may be to use `uuid.uuid4()`.

Reply:
There are no crytographic considerations here. The random number is only used to ensure that two pools have different names. So the only requirement is that success calls to random() deliver N distinct numbers where N is the number of LDAP servers being contacted. It seems likely that N will be < 10 and random() is easily able to meet the requirement.